core/src/storage.rs

```diff
@@ -173,12 +173,12 @@ impl InMemoryWaitList {
 }
 
 impl WaitList for InMemoryWaitList {
-    fn insert(&mut self, waker_id: MessageId, message: Message) {
-        self.inner.insert(waker_id, message);
+    fn insert(&mut self, id: MessageId, message: Message) {
```

Comment:
`WaitList` should be partitioned by `program_id`

Reply:
Ok, I'll add an issue.